pkg/controller/daemon/daemoncontroller.go

```diff
@@ -520,6 +527,51 @@ func (dsc *DaemonSetsController) updatePod(old, cur interface{}) {
 	}
 }
 
+func (dsc *DaemonSetsController) listSuspendedDaemonPods(node string) (dss []string) {
+	dsc.suspendedDaemonPodsLock.Lock()
+	defer dsc.suspendedDaemonPodsLock.Unlock()
+
+	if _, found := dsc.suspendedDaemonPods[node]; !found {
+		return []string{}
+	}
+
+	for k := range dsc.suspendedDaemonPods[node] {
+		dss = append(dss, k)
+	}
+	return
+}
+
+func (dsc *DaemonSetsController) requeueSuspendedDaemonPods(node string) {
+	dss := dsc.listSuspendedDaemonPods(node)
+	for _, dsKey := range dss {
+		if ns, name, err := cache.SplitMetaNamespaceKey(dsKey); err == nil {
+			if ds, err := dsc.dsLister.DaemonSets(ns).Get(name); err == nil {
+				dsc.enqueueDaemonSet(ds)
+			}
+		}
+	}
+}
+
+func (dsc *DaemonSetsController) addSuspendedDaemonPods(node, ds string) {
+	dsc.suspendedDaemonPodsLock.Lock()
+	defer dsc.suspendedDaemonPodsLock.Unlock()
+
+	if _, found := dsc.suspendedDaemonPods[node]; !found {
+		dsc.suspendedDaemonPods[node] = sets.NewString()
+	}
+	dsc.suspendedDaemonPods[node].Insert(ds)
+}
+
+func (dsc *DaemonSetsController) removeSuspendedDaemonPods(node, ds string) {
+	dsc.suspendedDaemonPodsLock.Lock()
+	defer dsc.suspendedDaemonPodsLock.Unlock()
+
+	if _, found := dsc.suspendedDaemonPods[node]; !found {
+		dsc.suspendedDaemonPods[node] = sets.NewString()
+	}
+	dsc.suspendedDaemonPods[node].Delete(ds)
```

Comment:
After this, I think you need to check if the corresponding Set is empty and remove it. Any system providing cluster auto-scaling may have a significant amount of Node churn. If you don't cleanup your map it could grow without bound.